typings/index.d.ts

```diff
@@ -278,23 +278,25 @@ export class Channel extends Base {
   public toString(): ChannelMention;
 }
 
-export class Client extends BaseClient {
+type If<T extends boolean, A, B = null> = T extends true ? A : T extends false ? B : A | B;
+
+export class Client<Ready extends boolean = boolean> extends BaseClient {
```

Comment:
why `boolean = boolean`?

Reply:
Defaulting to `boolean` if the generic isn't specified so as to not break existing typings.